cryostatio/cryostat-web
# fix(event-templates): should update template selection when template list updates

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes: #1002 
## Description of the change:

Whenever the template list changes, check with current template selection. If contained, do nothing. Otherwise, remove the selection.

Also, move `TargetSelect` and `SelectTemplateSelectorForm` to `@app/Shared` since they are pretty by themselves in their directory.

## Motivation for the change:

See #1002 

## How to manually test:
1. Run `CRYOSTAT_IMAGE=quay.io... sh smoketest.sh...`
2. Go to Rules. Click Create.
3. Enter a match expression that matches some targets. Select a template. Now, enter another match expression.
4. Notice the template selection will update as expected (not contained -> reset; contained -> do nothing).
